client/src/main/java/org/asynchttpclient/netty/channel/ChannelManager.java

```diff
@@ -490,7 +491,7 @@ public EventLoopGroup getEventLoopGroup() {
   }
 
   public ClientStats getClientStats() {
-    Map<String, Long> totalConnectionsPerHost = openChannels.stream().map(Channel::remoteAddress).filter(a -> a.getClass() == InetSocketAddress.class)
+    Map<String, Long> totalConnectionsPerHost = openChannels.stream().map(Channel::remoteAddress).filter(Objects::nonNull).filter(a -> a.getClass() == InetSocketAddress.class)
```

Comment:
Actually, there are 2 issues: you could be using one single filter, and (existing issue), equality comparison shouldn't be using ==.

`filter(a -> a instanceof InetSocketAddress)`